File: src/service/types/tag.ts
import { OrderFilter } from '@Utils/Order';

export enum TagKind {
  PRODUCT_TYPE = 'PRODUCT_TYPE',
}

interface Tag {
  id: string;
  name: string;
  kind: TagKind;
  color: string;
}

interface CreateTagInput {
  name: string;
  kind: TagKind;
  color: string;
}

interface ISearchTagsInput {
  kind: TagKind;

  page: number;
  per_page: number;
  order: OrderFilter;
}

type SearchTagsInput = Partial<ISearchTagsInput>;

interface TagsSearchResponse {
  total: number;
  skus: Tag[];
}

export type { Tag, CreateTagInput, SearchTagsInput, TagsSearchResponse };

File: src-tauri/src/ipc/input/tag.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Deserialize, Serialize, Clone)]
pub struct CreateTagInput {
    pub name: String,
    pub kind: String,
    pub color: String,
}
#[derive(Debug, Deserialize, Serialize, Clone)]
pub struct SearchTagsInput {
    pub kind: Option<String>,

    pub page: Option<u32>,
    pub per_page: Option<u32>,
    pub order: Option<String>,
}
